### PR TITLE
Change for NodeConfiguration build version feature

### PR DESCRIPTION
**Pull request by vrdmr**
## Description
The changes are generated based on the Hydra spec changes done (Commit: https://github.com/Azure/hydra-specs-pr/pull/1397). This adds support for new switch flag to be passed to signal incrementing the build number.

PS: Also adding the change in the spec in Swagger as well. That work is in progress.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/ResourceManagement/Automation/AutomationManagement/Microsoft.Azure.Management.Automation.nuget.proj [hunk 1]
```diff
@@ -5,7 +5,7 @@
     Microsoft.Azure.Management.Automation
     -->
     <SdkNuGetPackage Include="Microsoft.Azure.Management.Automation">
-      <PackageVersion>2.0.1</PackageVersion>
+      <PackageVersion>2.0.2</PackageVersion>
```
Comment (safeermohammed) [inline src/ResourceManagement/Automation/AutomationManagement/Microsoft.Azure.Management.Automation.nuget.proj]: Nuget already seems to have 2.0.3. Make it 2.0.4 if this is conflicting

File: src/ResourceManagement/Automation/AutomationManagement/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -20,7 +20,7 @@
 [assembly: AssemblyDescription("Provides Microsoft Azure Automation management operations.")]
 
 [assembly: AssemblyVersion("2.0.0.0")]
-[assembly: AssemblyFileVersion("2.0.1.0")]
+[assembly: AssemblyFileVersion("2.0.2.0")]
```
Comment (safeermohammed) [inline src/ResourceManagement/Automation/AutomationManagement/Properties/AssemblyInfo.cs]: Same here

**Review by safeermohammed — APPROVED** [review]

File: src/ResourceManagement/Automation/AutomationManagement/Microsoft.Azure.Management.Automation.nuget.proj (hunk 1 above)
Comment (vrdmr) [inline src/ResourceManagement/Automation/AutomationManagement/Microsoft.Azure.Management.Automation.nuget.proj]: Okay. I'll update it to 2.0.4 to be safe.

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

File: src/ResourceManagement/Automation/AutomationManagement/Microsoft.Azure.Management.Automation.nuget.proj [hunk 3]
```diff
@@ -5,7 +5,7 @@
     Microsoft.Azure.Management.Automation
     -->
     <SdkNuGetPackage Include="Microsoft.Azure.Management.Automation">
-      <PackageVersion>2.0.2</PackageVersion>
+      <PackageVersion>2.0.4</PackageVersion>
```
Comment (shahabhijeet) [inline src/ResourceManagement/Automation/AutomationManagement/Microsoft.Azure.Management.Automation.nuget.proj]: @vrdmr why skip 2.0.3?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@vrdmr add link to hyak spec PR

File: src/ResourceManagement/Automation/AutomationManagement/Microsoft.Azure.Management.Automation.nuget.proj (hunk 3 above)
Comment (vrdmr) [inline src/ResourceManagement/Automation/AutomationManagement/Microsoft.Azure.Management.Automation.nuget.proj]: @safeermohammed added a comment to the earlier commit - "Nuget already seems to have 2.0.3. Make it 2.0.4 if this is conflicting". I verified in the Nuget package manager to verify that a 2.0.3 version was published for the Automation.dll, which was released on 3/23/2017. We couldn't find out how was this released (if the code says 2.0.2 is the latest). Thus, to be on a safer side, we bumped it to 2.0.4.

**Comment by vrdmr** [comment]
Hyak specs PR - https://github.com/Azure/hydra-specs-pr/pull/1397

@shahabhijeet Could you please review this PR? Thanks.

File: src/ResourceManagement/Automation/AutomationManagement/Microsoft.Azure.Management.Automation.nuget.proj (hunk 3 above)
Comment (safeermohammed) [inline src/ResourceManagement/Automation/AutomationManagement/Microsoft.Azure.Management.Automation.nuget.proj]: @shahabhijeet  

We see 2.0.3 is already released to nuget. AFAIK we did not release this (Also from Azure repo history 2.0.2 was the last version). 

Can we track down how 2.0.3 was released and by who? 

https://www.nuget.org/packages/Microsoft.Azure.Management.Automation/

Comment (shahabhijeet) [inline src/ResourceManagement/Automation/AutomationManagement/Microsoft.Azure.Management.Automation.nuget.proj]: @vrdmr @safeermohammed the only option is to see the nuget.org release date and track it by finding the publish jobs that were run around that time.
We are trying to put in more meta data that will allow us to track this much better in the future.

**Comment by shahabhijeet** [comment]
@vrdmr when is Automation moving/updating Swagger/AutoRest version?
We would like to know the plan moving forward?

**Comment by vrdmr** [comment]
@shahabhijeet I have started writing the test-suite for SDK generated by AutoRest branch (Swagger). I transitioned it from @vivlingaiah recently and would be taking it forward. For my changes for NodeConfiguration build versioning, I made changes in both the places. Others in the team are following the suite as well.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**